File: OneDrive/Desktop/EventBookingApp/src/main/java/com/ranadj/service/PackageService.java
package com.ranadj.service;

import com.ranadj.dto.PackageRequest;
import com.ranadj.dto.PackageResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

/**
 * Service interface defining package management business logic.
 */
public interface PackageService {

    /**
     * Creates a new package.
     */
    PackageResponse createPackage(PackageRequest request);

    /**
     * Updates an existing package.
     */
    PackageResponse updatePackage(Long id, PackageRequest request);

    /**
     * Deletes (purges) a package by its ID.
     */
    void deletePackage(Long id);

    /**
     * Retrieves a package by its ID.
     */
    PackageResponse getPackageById(Long id);

    /**
     * Retrieves all packages with advanced pagination support.
     * Optionally filters to only active packages.
     */
    Page<PackageResponse> getAllPackages(Pageable pageable, Boolean activeOnly);
}

File: src/services/package.service.js
import api from './api'

const packageService = {
  // Public - Get all active packages
  getAllPackages: async () => {
    const response = await api.get('/packages')
    return response.data.data
  },

  // Admin - Get single package
  getPackageById: async (id) => {
    const response = await api.get(`/packages/${id}`)
    return response.data.data
  },

  // Admin - Create a new package
  createPackage: async (packageData) => {
    const response = await api.post('/packages', packageData)
    return response.data.data
  },

  // Admin - Update an existing package
  updatePackage: async (id, packageData) => {
    const response = await api.put(`/packages/${id}`, packageData)
    return response.data.data
  },

  // Admin - Delete a package
  deletePackage: async (id) => {
    const response = await api.delete(`/packages/${id}`)
    return response.data.data || response.data
  }
}

export default packageService
